src/stdune/bin.ml

```diff
@@ -40,7 +40,13 @@ let which ~path prog =
       let fn = Path.relative dir prog in
       Option.some_if (exists fn) fn)
 
-let make ~path =
+let unix_make ~path =
   match which ~path "gmake" with
   | None -> which ~path "make"
   | some -> some
+
+let make =
+  if Sys.unix then
+    unix_make
+  else
+    which "make"
```

Comment:
```suggestion
let make ~path =
  let gmake = if Sys.unix then which ~path "gmake" else None in
  match gmake with
  | None -> which ~path "make"
  | some -> some
```

Reply:
This code becomes:
```ocaml
let make ~path =
  let gmake =
    if Sys.unix then
      which ~path "gmake"
    else
      None
  in
  match gmake with
  | None -> which ~path "make"
  | some -> some
```
after formatting with `ocamlformat`. Is that still the preferred version?

Reply:
I originally split this code like this because it allows to only execute the condition once regardless of whether the compiler will optimize it or not.

Reply:
I still find the suggested version more readable, but it is subjective. Let's give some time for the some other developers to chime in. Thanks!

Reply:
I like that `which ~path "gmake"` only occurs once in this version.

Reply:
Fair enough. Done!